packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -373,11 +632,64 @@ export class SankeyChartBase extends React.Component<ISankeyChartProps, ISankeyC
     };
     this._calloutId = getId('callout');
     this._linkId = getId('link');
+    this._chartId = getId('sankeyChart');
+    this._emptyChartId = getId('_SankeyChart_empty');
+    this._labelTooltipId = getId('tooltip');
     this._margins = { top: 36, right: 48, bottom: 32, left: 48 };
-    this._normalizeData = memoizeFunction((data: ISankeyChartData, containerWidth: number, containerHeight: number) =>
-      this._normalizeSankeyData(data, containerWidth, containerHeight),
+    // We memo-ize creation so that we only create a new object when any of the fields change.
+    this._computeClassNamesProps = memoizeFunction(
+      (
+        theme: ITheme,
+        pathColor: string,
+        className: string,
+        containerWidth: number,
+        containerHeight: number,
+      ): ISankeyChartStyleProps => ({
+        theme: theme!,
+        width: containerWidth,
+        height: containerHeight,
+        pathColor,
+        className,
+      }),
     );
-    this._emptyChartId = getId('_SankeyChart_empty');
+    // `getClassNames` is memoized underneath, so it only recomputes when the `styles` or `classNamesProps` change.
+    // We memoize `classNamesProps` so that we only create new class names when absolutely necessary,
+    // and so that we can memoize the `_createNodes` method.
+    this._computeClassNames = memoizeFunction(
```

Comment:
>memoizeFunction [](http://example.com/codeflow?start=30&length=15)

ideally we shouldn't need so much memoization. React is pretty good at memoizing internally as long as we are good a state management and not changing states too often. Which seems to be not very good for the sankey chart control.